src/Containers/DropArea/DropArea.tsx

```diff
@@ -0,0 +1,179 @@
+import { flex } from '@Utils/Mixins';
+import Dropzone, {
+    useDropzone,
+    DropEvent,
+    FileRejection,
+} from 'react-dropzone';
+import Lottie from 'react-lottie';
+import { animationData } from './animationData';
+
+const lottieOptions = {
+    loop: true,
+    autoplay: true,
+    animationData,
+    rendererSettings: {
+        preserveAspectRatio: 'xMidYMid slice',
+    },
+};
+
+export interface IDropAreaProps
+    extends MainInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    message?: string;
+    buttonText?: string;
+    onClick?: React.MouseEventHandler<HTMLElement> | undefined;
+    onDragEnter?: React.DragEventHandler<HTMLElement> | undefined;
+    onDragLeave?: React.DragEventHandler<HTMLElement> | undefined;
+    onDropHandler?:
+        | (<T extends File>(
+              acceptedFiles: T[],
+              fileRejections: FileRejection[],
+              event: DropEvent,
+          ) => void)
+        | undefined;
+    isDisabled?: boolean;
+    width?: number;
+}
+
+export const DropArea: React.FC<IDropAreaProps> = ({
+    message = 'Drag & Drop your files here',
+    buttonText = 'Browse Files',
+    onClick = () => null,
+    onDragEnter = () => null,
+    onDragLeave = () => null,
+    onDropHandler = () => null,
+    isDisabled = false,
+    ...props
+}): React.ReactElement => {
+    const { getInputProps, getRootProps, isDragActive } = useDropzone({
+        onDragEnter,
+        onDragLeave,
+        onDrop: onDropHandler,
+        disabled: isDisabled,
+    });
+
+    const getLottieAnimationOrIcon = (isDragEnter: boolean): JSX.Element => {
+        if (isDragEnter)
+            return <Lottie options={lottieOptions} width={140} height={60} />;
+        return <Icon as={CloudUploadAlt} />;
+    };
+
+    const stopPropagation = (e: React.MouseEvent): void => {
+        e.stopPropagation();
+    };
+
+    return (
+        <Dropzone multiple>
+            {() => (
+                <RootDiv {...getRootProps({})} onClick={stopPropagation}>
+                    <DropAreaBox isDragEnter={isDragActive} {...props}>
+                        {getLottieAnimationOrIcon(isDragActive)}
+                        <MessageBox>{message}</MessageBox>
+                        <OrBox>OR</OrBox>
+                        <BrowseFiles>
+                            {buttonText}
+                            <Input
+                                {...getInputProps({
+                                    onClick,
+                                    disabled: isDisabled,
+                                })}
+                            />
+                        </BrowseFiles>
+                    </DropAreaBox>
+                </RootDiv>
+            )}
+        </Dropzone>
+    );
+};
+
+const RootDiv = styled.div`
+    width: fit-content;
+`;
+
+const Input = styled.input`
+    opacity: 0;
+    display: initial !important;
+    visibility: initial;
+    position: absolute;
+    z-index: 99999999;
```

Comment:
This is all not needed. Put pointer Event to pass through all parents or whatever is capturing the event and then you can remove this hack

Reply:
but then the dragging behaviour doesn't work

Reply:
It will work, dragging behaviour is not affected by this

Reply:
yes, it is! it stops working! I tell you. If I have done this change (go back to input styled component) it's because dragging behaviour was not working, and the reason for this was the CSS property pointer-events:none; I can tell because when I removed that property it started to work again the dragging behaviour. I think it's because the on-hover event, that is not detected when pointer-events:none;

Reply:
I will try some things but I suppose I will not succeed...

Reply:
I tried and did not succeed. The only way to achieve the desired result is as it is now (with the hacky CSS).

Reply:
Use a different property other than none. Like pass through . None will not work

Reply:
there not exist this property for pointer-events

Reply:
the thing is when the click event reaches RootDiv, it triggers onclick event on input element. we don't want this behavior. it can be canceled by assigning an onClick handler to the RootDiv, as I have done assigning stopPropagation function.

Reply:
maybe I can avoid the hacky css by using a reference to the input element and triggering the click event in the onclick handler of the browseFiles element

Reply:
I can do without the hacky css but paying a price, the price is you cannot preventDefault behaviour, so this means you cannot prevent from the dialog box to select a file to appear (that's why it's important to test and hence the importance of the story you told me to delete). do you want me to keep the hacky css and being able to prevent default behaviour or you want to loose this ability and have the code without the hacky css?

Reply:
You can prevent at the function layer

Reply:
Just do it the correct way I will show you how, you call some code before you call the omClick event

Reply:
https://stackoverflow.com/q/66621550/12441000

Reply:
when you dispatch an event programmatically you cannot prevent default behavior